jetty-jaspi/src/main/java/org/eclipse/jetty/security/jaspi/JaspiAuthenticatorFactory.java

```diff
@@ -140,13 +157,13 @@ protected Subject findServiceSubject(Server server)
     }
 
     /**
-     * Find a servername.
-     * If {@link #setServerName(String)} has not been called, then
-     * use the name of the a principal in the service subject.
-     * If not found, return "server".
+     * Find a servername. If {@link #setServerName(String)} has not been called,
+     * then use the name of the a principal in the service subject. If not found,
+     * return "server".
      *
      * @param server the server to find the name of
-     * @return the server name from the service Subject (or default value if not found in subject or principals)
+     * @return the server name from the service Subject (or default value if not
+     *         found in subject or principals)
```

Comment:
According to the spec, this should be the value returned from `ServletContext.getVirtualServerName()`.  In jetty's case, this will be the first `virtual host` for the context, or `null` if there aren't any.  So I think you should ask the context for it's virtualServerName, and if you get `null`, fall back to these other ways of obtaining it, finally defaulting to the string `server`.

NB you will also have to update the jaspi-authmoduleconfig.xml file comment to reflect the above algorithm.

Reply:
the logic in this class hasn't been changed, but I've thrown in a call to the ServletContext.getVirtualServerName to try that if the _serverName member variable hasn't been configured.

I'm not entirely sure if the existing logic of using the first Subject's first principal name in the original logic is appropriate, but kept it as it is current behaviour.